GitUI/UserControls/RevisionGrid.cs

```diff
@@ -2080,7 +2080,11 @@ private void RevisionsCellFormatting(object sender, DataGridViewCellFormattingEv
                 if (revision.Body == null && !revision.IsArtificial)
                 {
                     var moduleRef = Module;
-                    ThreadPool.QueueUserWorkItem(o => LoadIsMultilineMessageInfo(revision, columnIndex, e.RowIndex, Revisions.RowCount, moduleRef));
```

Comment:
This significantly slows down the revision grid scrolling. Select a revision and keep the arrow down button pressed.
Compare it to the current master. Is there any good practice about scheduling task with lower priority? (dedicated TaskScheduler?)

Reply:
💭 I'm not seeing a big difference between c6dfc3c03c0fbc9ec562261b456d4286d6860396 and 9a1f35487761e1e773bafbdb0dadfe627217490c. Are you using this repository as the sample to demonstrate the issue?

Reply:
Try turning on the "show long commit message indicator" option in the view menu. I know @jbialobr mentioned that setting when I looked at threading changes before. It adds a lot more load during scrolling.

Reply:
Thanks @drewnoakes
![gitext](https://user-images.githubusercontent.com/1003909/38043131-86b6e4c4-32b6-11e8-9920-b0ef235cdfa5.png)

Reply:
@jbialobr I'm still not observing a major *difference* compared to before. From what I can tell in code, this seems to be a longstanding problem, where multiple features are inefficiently interacting with git.

Reply:
> From what I can tell in code, this seems to be a longstanding problem, where multiple features are inefficiently interacting with git.

I believe that applying non blocking async calls all the way down will slightly improve perf here. For  now I would prefer leaving LoadIsMultilineMessageInfo as it was and come back to it after getting async top to bottom done.

Try also keeping the page down button pressed.

Reply:
@jbialobr To be clear, the performance issue is very obvious. I can easily encounter situations with 10+ second delays. However, I can hit these delays in exactly the same manner before any of my changes are applied. The differences I was able to locate were in the range of ~2% of the total delays, which doesn't seem big enough to block this.

Reply:
Without your changes, scrolling revisions with arrow down/page down is pretty fluent. After this change the scrolling blocks frequently and is no more fluent. The UX is poor.